src/ert/_c_wrappers/enkf/site_config.py

```diff
@@ -1,59 +1,119 @@
-    _get_config_file = ResPrototype("char* site_config_get_config_file(site_config)")
-    _get_env_var_list = ResPrototype(
-        "env_varlist_ref site_config_get_env_varlist(site_config)"
-    )
-
-    def __init__(self, config_content=None):
-
-        if config_content:
-            c_ptr = self._alloc(config_content)
-        else:
-            c_ptr = self._site_config_alloc_default()
-
-        if c_ptr is None:
-            raise ValueError("Failed to construct SiteConfig instance.")
-
-        super().__init__(c_ptr)
+from ert._c_wrappers.job_queue import EnvironmentVarlist, ExtJob, ExtJoblist
+
+
+@dataclass
+class SiteConfig:
+    job_list: ExtJoblist
+    var_list: EnvironmentVarlist
+
+    @classmethod
+    def _add_config_content(
+        cls,
+        license_root_path: str,
+        site_config_content: ConfigContent,
+        job_list: ExtJoblist,
+        var_list: EnvironmentVarlist,
+    ):
+
+        if site_config_content.hasKey(ConfigKeys.INSTALL_JOB):
+            for args in iter(site_config_content[ConfigKeys.INSTALL_JOB]):
+                if not os.path.isfile(args[1]):
+                    print(f"WARNING: Unable to locate job file {args}")
+                    continue
+                try:
+                    new_job = ExtJob(
+                        config_file=args[1],
+                        private=False,
+                        name=args[0],
+                        license_root_path=license_root_path,
+                    )
+                    new_job.convertToCReference(None)
+                    job_list.add_job(new_job.name(), new_job)
+                except (ValueError, OSError):
+                    print(f"WARNING: Unable to create job from {args}")
+
+        if site_config_content.hasKey(ConfigKeys.INSTALL_JOB_DIRECTORY):
+            for args in iter(site_config_content[ConfigKeys.INSTALL_JOB_DIRECTORY]):
+                job_path = args[0]
+                if not os.path.isdir(job_path):
+                    print(f"WARNING: Unable to locate job directory {job_path}")
+                    continue
+                files = os.listdir(job_path)
+                for file_name in files:
+                    full_path = os.path.join(job_path, file_name)
+                    if os.path.isfile(full_path):
+                        try:
+                            new_job = ExtJob(
+                                config_file=full_path,
+                                private=False,
+                                license_root_path=license_root_path,
+                            )
+                            new_job.convertToCReference(None)
+                            job_list.add_job(new_job.name(), new_job)
+                        except (ValueError, OSError):
+                            print(
+                                f"WARNING: Unable to create job from {full_path} "
+                                f"in directory {job_path}"
+                            )
```

Comment:
and again here, i think, you could perhaps refactor the two blocks that add a job, like i commented below line 151 (sorry i didn't read linearly)

Reply:
I think I was a bit rash in saying it was too different, so I think I can make one function to rule them all.